lighthouse/routes/common/beckman.py

```diff
@@ -1,21 +1,14 @@
 import logging
 
-from flask import Blueprint
-from flask_cors import CORS
-
 from lighthouse.classes.beckman import Beckman
 from lighthouse.constants.error_messages import ERROR_UNEXPECTED
 from lighthouse.helpers.responses import internal_server_error, ok
 from lighthouse.types import FlaskResponse
 
 logger = logging.getLogger(__name__)
 
-bp = Blueprint("beckman", __name__)
-CORS(bp)
-
 
-@bp.get("/beckman/robots")
-def get_robots() -> FlaskResponse:
+def get_robots_v1() -> FlaskResponse:
```

Comment:
I might need some context but this is a common file but it has `_v1` in the function name which seems contradictory.

Reply:
I wasn't sure how to address this.  I can remove the `v1` naming of functions if you like, but think about what happens when `v2` comes out.  I think you still want `v2` to serve up `v1` endpoints that aren't changed.  However where an endpoint has changed between `v1` and `v2`, the two versions should serve different functionality as appropriate.  Adding these numbered methods would mean you can create one more for a `v2` endpoint, then the blueprint for `v2` endpoints could send the request to one method or another depending on whether it's changed or not.  It would be OK for a `v2` endpoint to point at the `v1` method if it hasn't changed.

The numbering isn't essential as you could still create additional methods, but the number indicates the introduction of the method at a glance.  If there's no method for `v2`, the blueprint should still point at `v1`.  When a blueprint for `v3` is introduced, some of the endpoints might be pointing at the `v1` implementation, at least one is probably pointing at a `v2` method and there will be at least one `v3` method to point at for another endpoint.

Reply:
I think it would be more appropriate to add the "date of addition" in comments. My understanding is that these are common methods that numerous versions can call and the logic remains the same, regardless of the calling version. The version specific logic should be in the versioned packages. Our structure might not align to the example in SO but I think that's the main idea.

Reply:
OK well I'll change it to that, but we can't be sure it remains as clean as that.  At the moment, the entire implementation is in this method for v1 and if v2 is basically the same, but something different has to happen in the middle of the process, then you'll have to modify this method in to two methods anyway to suit being used for v1 and v2.  And you'd have to call both common methods from a v1 specific and a v2 specific method.  I think overall it's much cleaner to have a version specific method in common.  The point being that in v2, only one of a dozen endpoints might change and so you only have to duplicate one method for v2 while the other v2 endpoints still call the common v1 implementation.

I think the reality here is that we don't know how the versioning might work until it happens.  Whatever we do today is probably not the optimum.

Reply:
Yip, I agree, this is the first stab at it so let's call it a day and see how things evolve!